### DISCUSSION TITLE
Best practices for parameter storage?

### DISCUSSION THREAD

**Discussion by ccrome** [adafruit/tinyuf2#290] [General]
What are the best practices for storing parameter information that can be shared between the bootloader and application?

For example, some typical stuff I'd like to store are:

**Fixed, unchangeable data.**
* board build date
* build factory
* serial number

**Variable Data** 
* board settings and/or data that might be collected at runtime.

**Comment by hathach** [comment]
uf2 has a specs for this called cf2: https://github.com/microsoft/uf2/blob/master/cf2.md . I usedd it a bit in nrf52 bootloader but haven't done so for tinyuf2 just yet https://github.com/adafruit/Adafruit_nRF52_Bootloader/blob/master/src/boards/feather_nrf52840_express/pinconfig.c
